Automattic/wp-calypso
# Theme Showcase: Hide Try & Customize action for block-based themes

#### Changes proposed in this Pull Request

* Themes that primarily use blocks will be underserved by trying to use the Customizer to set up their site; we'd rather they use the editor instead.
* We currently hide Try & Customize for block-based themes in the theme's three-dots menu, but it appears as a secondary action in the Live demo window
* This PR moves the logic for hiding the Try & Customize action into a selector so we can reuse it.
* Check `secondaryOption.hideForTheme( themeId, siteId )` before outputting it to the live preview on the main Theme Showcase page.
* Check the new selector before outputting the Try & Customize action to the theme preview on individual theme sheets.

**Before**

<img width="1308" alt="Screen Shot 2022-01-19 at 10 16 40 AM" src="https://user-images.githubusercontent.com/2124984/150159731-9bcec593-19b4-4d95-a559-901ff983703d.png">

**After**

<img width="1311" alt="Screen Shot 2022-01-19 at 10 13 17 AM" src="https://user-images.githubusercontent.com/2124984/150160538-e5085d9f-a2f2-422f-9bcf-7d660d4836da.png">

#### Testing instructions

* Switch to this PR, navigate to `/themes/[site]`
* Any theme that has a Try & Customize option in their three-dots menu (Karuna, for example) should still have a Try & Customize secondary action when you view the theme's Live demo.
* Likewise, any theme that does not have a Try & Customize option in their three-dots menu (Byrne, for example) should *not* have a Try & Customize secondary action when you view the theme's Live demo.
* You should get the same results when viewing individual themes' pages and clicking "Open live demo"
* You should not be able to Try & Customize when you're viewing the logged out Showcase

Related to #59991
